cmd/aws-vpc-cni/main.go

```diff
@@ -292,23 +295,41 @@ func generateJSON(jsonFile string, outFile string, getPrimaryIP func(ipv4 bool)
 
 	byteValue = []byte(netconf)
 
+	// Unmarshall current conflist into data in order to chain any requested plugins
+	data := NetConfList{}
+	err = json.Unmarshal(byteValue, &data)
+	if err != nil {
+		return err
+	}
+
+	// Chain the bandwidth plugin when enabled
 	enBandwidthPlugin := utils.GetBoolAsStringEnvVar(envEnBandwidthPlugin, defaultEnBandwidthPlugin)
 	if enBandwidthPlugin {
-		data := NetConfList{}
-		err = json.Unmarshal(byteValue, &data)
-		if err != nil {
-			return err
-		}
-
 		bwPlugin := NetConf{
 			Type:         "bandwidth",
 			Capabilities: map[string]bool{"bandwidth": true},
 		}
 		data.Plugins = append(data.Plugins, &bwPlugin)
-		byteValue, err = json.MarshalIndent(data, "", "  ")
-		if err != nil {
-			return err
+	}
+
+	// Chain the tuning plugin (configured to disable IPv6 in container network namespace) when requested
+	disableContainerV6 := utils.GetBoolAsStringEnvVar(envDisableContainerV6, defaultDisableContainerV6)
+	if disableContainerV6 {
+		tuningPlugin := NetConf{
+			Type: "tuning",
+			Sysctl: map[string]string{
+				"net.ipv6.conf.all.disable_ipv6":     "1",
+				"net.ipv6.conf.default.disable_ipv6": "1",
+				"net.ipv6.conf.lo.disable_ipv6":      "1",
```

Comment:
Not in any of my testing, but since the loopback interface is already created and assigned an IPv6 address by the time this chained plugin runs, I figured it was best to be safe and not make any assumptions on how the container kernel will handle cleanup.

Reply:
Yeah, we wouldn't need this. All the containers share the Kernel with the host machine/OS and we're only doing this per network ns. But there is no harm doing this, so we can leave it.